app/scripts/lib/state-corruption-errors.ts

```diff
@@ -0,0 +1,97 @@
+import log from 'loglevel';
+import { escape as lodashEscape } from 'lodash';
+import {
+  SUPPORT_LINK,
+  VAULT_RECOVERY_LINK,
+} from '../../../shared/lib/ui-utils';
+import {
+  getErrorHtmlBase,
+  getLocaleContext,
+  setupLocale,
+} from '../../../shared/lib/error-utils';
+import {
+  MISSING_VAULT_ERROR,
+  CORRUPTION_BLOCK_CHECKSUM_MISMATCH,
+} from '../../../shared/constants/errors';
+import { switchDirectionForPreferredLocale } from '../../../shared/lib/switch-direction';
+import getFirstPreferredLangCode from './get-first-preferred-lang-code';
+
+export type ErrorLike = {
+  message: string;
+  name: string;
+  stack?: string;
+};
+
+export const METHOD_DISPLAY_STATE_CORRUPTION_ERROR =
+  'displayStateCorruptionError';
+
+export const KNOWN_STATE_CORRUPTION_ERRORS = new Set([
+  MISSING_VAULT_ERROR,
+  CORRUPTION_BLOCK_CHECKSUM_MISMATCH,
+]);
+
+export async function getStateCorruptionErrorHtml(
+  vaultRecoveryLink: string,
+  currentLocale?: string,
+  supportLink?: string,
+) {
+  let response, preferredLocale: string;
+  if (currentLocale) {
+    preferredLocale = currentLocale;
+    response = await setupLocale(currentLocale);
+  } else {
+    preferredLocale = await getFirstPreferredLangCode();
+    response = await setupLocale(preferredLocale);
+  }
+
+  switchDirectionForPreferredLocale(preferredLocale);
+  const { currentLocaleMessages, enLocaleMessages } = response;
+  const t = getLocaleContext(currentLocaleMessages, enLocaleMessages);
+
+  const header = `<p>
```

Comment:
[non blocker] Not sure if there's a mockup for this screen, but there is one around error screen for page crash, do you want to reuse some bits to fill the gap of UI and have a consistent style ?
<img width="400" alt="Screenshot 2025-04-19 at 00 34 09" src="https://github.com/user-attachments/assets/f7f260c0-91ec-4eab-8cfe-72fec03edb0a" />

Reply:
Since there is duplication in the state corruption and our existing critical error UI code, i'll lump this design work in with the refactoring work. I've added it to the epic for vault recovery, you can track the PR here: https://github.com/MetaMask/metamask-extension/pull/32160

Reply:
Sounds good, thanks for the work David 👍🏻